tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/util/ServerUtil.java

```diff
@@ -487,23 +488,58 @@ public static PortalBundle getPortalBundle(IProject project)  throws CoreExcepti
 
         IStatus status = sdk.validate();
 
-        if ( !status.isOK())
+        if ( !status.isOK() )
         {
             return null;
         }
 
+        final String cachedAppServerPath = LiferayServerCore.getPreference(sdk.getName() + ".server.path");
```

Comment:
while the cacahe appServerPath or type wil be null or it isn't same as the latest build properties file, then the cached appServerPath will be override.